run/run_summary.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class TestRunSpec(object):
+    def __init__(self, sha, platform):  # type: (str, str) -> None
+        self.sha = sha
+        self.platform = platform
+
+    @classmethod
+    def parse(cls, value):  # type: (str) -> TestRunSpec
+        pieces = value.split('@')
+        sha = 'latest'
+        if len(pieces) > 2 or len(pieces) < 2:
```

Comment:
Just sha, platform = pieces should replace a number of lines here, it'll throw ValueError if the array isn't of size 2.